content/blog/2018-03-27-update-on-async-rendering.md

```diff
@@ -128,6 +129,12 @@ Although the above code is not problematic in itself, the `componentWillReceiveP
 As of version 16.3, the recommended way to update `state` in response to `props` changes is with the new `static getDerivedStateFromProps` lifecycle. (That lifecycle is called when a component is created and each time it receives new props):
 `embed:update-on-async-rendering/updating-state-from-props-after.js`
 
+You may notice in the example above that `props.currentRow` is mirrored in state (as `state.lastRow`). This enables `getDerivedStateFromProps` to access the previous props value in the same way as is done in `componentWillReceiveProps`.
```

Comment:
I was also wondering while reading this if `getDerivedStateFromProps` needs to return an object of `type State` or `$Subtype<State>`

Reply:
I originally included Flow annotations for the new lifecycle but we got feedback that they were confusing.

The return type for `getDerivedStateFromProps` is `$Shape<State>`

Reply:
(It is the exact same type as supported by `setState`)

Reply:
@iamdustan There's a link to type definitions from the blog post. https://gist.github.com/gaearon/88634d27abbc4feeb40a698f760f3264

Reply:
👍 I assumed that it was the same type as `setState` but couldn’t remember if I remembered the RFC correctly. 

I missed that gist link while reading through that introduction-to-methods section. Thanks fellows.